book_source/02_demos_tutorials_workflows/04_more_web_interface/02_hidden_analyses.Rmd

```diff
@@ -292,8 +292,48 @@ where we calculate the mean of the process covariance matrix $\left(\bar{\boldsy
 Users have control over how they think is the best way to estimate $Q$. Our code will look for the tag `q.type` in the XML settings under `state.data.assimilation` which can take 3 values of Single, PFT or Site. If `q.type` is set to single then one value of process variance will be estimated across all different sites or PFTs. On the other hand, when q.type` is set to Site or PFT then a process variance will be estimated for each site or PFT at a cost of more time and computation power.
 
 #### Multi-site State data assimilation.
+`sda.enkf.multisite` is housed within: `/pecan/modules/assim.sequential/R`
+
+The 4-site tutorial is housed within: `~/pecan/modules/assim.sequential/inst/MultiSite-Exs`
+
+#### **sda.enkf.multisite.R Description**
 `sda.enkf.multisite` function allows for assimilation of observed data at multiple sites at the same time. In order to run a multi-site SDA, one needs to send a multisettings pecan xml file to this function. This multisettings xml file needs to contain information required for running at least two sites under `run` tag. The code will automatically run the ensembles for all the sites and reformats the outputs matching the required formats for analysis step. 
+#### **sda.enkf.multisite.R Arguments**
+* settings - (required) [State Data Assimilation Tags Example] settings object
+
+* obs.mean - (required) List of sites named by site ids, which contains dataframe for observation means, named with observation datetime.
+
+* obs.cov - (required) List of sites named by site ids, which contains dataframe for observation covariance matrix, named with observation datetime.
+
+* Q - (optional) Process covariance matrix given if there is no data to estimate it. Defualt is NULL.
+
+* restart - (optional) Used for iterative updating previous forecasts. Default NULL. List object includes file path to previous runs and start date for SDA. Defualt is NULL.
+
+* pre_enkf_params - (optional) Used for carrying out SDA with pre-existed enkf.params, in which the Pf, aqq, and bqq can be used for the analysis step. Defualt is NULL.
+
+* ensemble.samples - (optional) Pass ensemble.samples from outside to avoid GitHub check issues. Defualt is NULL.
 
+* control - (optional) List of flags controlling the behaviour of the SDA. Default is as follows:
+```
+control=list(trace = TRUE,
+             FF = FALSE,
+             interactivePlot = FALSE,
+             TimeseriesPlot = FALSE,
+             BiasPlot = FALSE,
+             plot.title = NULL,
+             facet.plots = FALSE,
+             debug = FALSE,
+             pause = FALSE,
+             Profiling = FALSE,
+             OutlierDetection=FALSE,
+             parallel_qsub = TRUE,
+             free_run = FALSE,
```

Comment:
So we probably need to remove "free_run" here? And "SDA_runner.R" should be updated too.

Reply:
Fixed.